spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/SparkWriteConf.java

```diff
@@ -708,4 +709,15 @@ private long sparkAdvisoryPartitionSize() {
   private double shuffleCompressionRatio(FileFormat outputFileFormat, String outputCodec) {
     return SparkCompressionUtil.shuffleCompressionRatio(spark, outputFileFormat, outputCodec);
   }
+
+  public DeleteGranularity deleteGranularity() {
```

Comment:
I did not add a SQL property cause I am not sure it makes sense at the session level.
Thoughts?

Reply:
Sounds reasonable to me. Config properties are more surface area to support.